### ISSUE TITLE
Fitting an AR-HMM only runs 1 iteration instead of 50 as specified 

### ISSUE THREAD

**Issue by juliagorman** [dattalab/keypoint-moseq#26]
Despite specifying 50 iterations of model fitting, model only runs one iteration and then moves on to the next line of code. This happens with different files I have tried. 
<img width="1262" alt="Screenshot 2023-04-25 at 11 30 42 AM" src="https://user-images.githubusercontent.com/32439305/234369397-029099d3-fe5f-4a94-a560-6b75af03a04c.png">

**Comment by calebweinreb** [comment]
Fitting terminated because NaNs were detected, as shown in the error message. The [troubleshooting docs](https://keypoint-moseq.readthedocs.io/en/latest/troubleshooting.html) describe some options for dealing with this.

**Comment by juliagorman** [comment]
Okay I imported as instructed above. The command still returned False, so I mangually set JAX as instructed as shown in my screenshot. Still I got the same error when training the AR-HMM and could run more than one iteration. Sorry for all these issues!

<img width="1130" alt="Screenshot 2023-04-26 at 11 18 36 AM" src="https://user-images.githubusercontent.com/32439305/234666860-654ebfdf-6bea-451d-be95-c2cdbff38405.png">

**Comment by juliagorman** [comment]
Also, I tried it in this order such that it then outputted True, and still I got this error with fitting.

<img width="934" alt="Screenshot 2023-04-26 at 11 46 25 AM" src="https://user-images.githubusercontent.com/32439305/234672940-834908ce-7a9f-4621-b390-d7c66563c31c.png">

<img width="1122" alt="Screenshot 2023-04-26 at 11 46 59 AM" src="https://user-images.githubusercontent.com/32439305/234673036-02d78d40-2e17-490a-892d-33c395682d0b.png">

**Comment by calebweinreb** [comment]
A couple thoughts:

1. Unless you're using google colab, I would recommend running the pip update commands outside the notebook, i.e. in another terminal window that has the `keypoint_moseq` conda environment activated. After that, you will need to restart the kernel of the notebook screenshotted above.

2. After updating the code, make sure you re-run the model fitting notebook from the beginning. i.e. don't use a saved checkpoint.

3. If you are using colab, then you should delete and disconnect the runtime, and just start over with a new runtime using the install commands as they appear in the [example colab](https://colab.research.google.com/github/dattalab/keypoint-moseq/blob/main/docs/keypoint_moseq_colab.ipynb). Also it seems from your screenshot like the GPU is not being used. The code will run much faster if you select GPU for the runtime type, as explained in the linked colab.

**Comment by juliagorman** [comment]
However, I do not run into this issue when running jupyter notebook so for the time being I have just switched to jupyter for the time being.

**Comment by juliagorman** [comment]
Oop I spoke to soon I actually start getting the same issue further down the code now and the applying the trained model step: 
<img width="1024" alt="Screenshot 2023-04-27 at 2 51 17 PM" src="https://user-images.githubusercontent.com/32439305/234998718-4d34ebb5-c3f3-49e2-af89-7c7fa121d172.png">

**Comment by calebweinreb** [comment]
I'll look into the jupyter notebook. For the issue above, see https://github.com/dattalab/keypoint-moseq/issues/15#issuecomment-1503626291

tldr; you can avoid this by setting "num_iters=0"
```
results = kpms.apply_model(coordinates=coordinates, confidences=confidences, 
                           project_dir=project_dir, **config(), **checkpoint, num_iters=0)
```
